tests/macros/stack.rs

```diff
@@ -0,0 +1,428 @@
+            let dyn_result = stack_dyn(
+                dmatrix![$($(stack_dyn_convert_to_dmatrix_option!($entry)),*);*]
+            );
+            // println!("{}", stack_result);
+            // println!("{}", dyn_result);
+            assert_eq!(stack_result, dyn_result);
+        }
+    }
+}
+
+#[test]
+fn stack_simple() {
+    let m = stack![
+        Matrix2::<usize>::identity(), 0;
+        0, &Matrix2::identity();
+    ];
+
+    assert_eq_and_type!(m, Matrix4::identity());
+}
+
+#[test]
+fn stack_diag() {
+    let m = stack![
+        0, matrix![1, 2; 3, 4;];
+        matrix![5, 6; 7, 8;], 0;
+    ];
+
+    let res = matrix![
+        0, 0, 1, 2;
+        0, 0, 3, 4;
+        5, 6, 0, 0;
+        7, 8, 0, 0;
+    ];
+
+    assert_eq_and_type!(m, res);
+}
+
+#[test]
+fn stack_dynamic() {
+    let m = stack![
+        matrix![ 1, 2; 3, 4; ], 0;
+        0, dmatrix![7, 8, 9; 10, 11, 12; ];
+    ];
+
+    let res = dmatrix![
+        1, 2, 0, 0, 0;
+        3, 4, 0, 0, 0;
+        0, 0, 7, 8, 9;
+        0, 0, 10, 11, 12;
+    ];
+
+    assert_eq_and_type!(m, res);
+}
+
+#[test]
+fn stack_nested() {
+    let m = stack![
+        stack![ matrix![1, 2; 3, 4;]; matrix![5, 6;]],
+        stack![ matrix![7;9;10;], matrix![11; 12; 13;] ];
+    ];
+
+    let res = matrix![
+        1, 2, 7, 11;
+        3, 4, 9, 12;
+        5, 6, 10, 13;
+    ];
+
+    assert_eq_and_type!(m, res);
+}
+
+#[test]
+fn stack_single() {
+    let a = matrix![1, 2; 3, 4];
+    let b = stack![a];
+
+    assert_eq_and_type!(a, b);
+}
+
+#[test]
+fn stack_single_row() {
+    let a = matrix![1, 2; 3, 4];
+    let m = stack![a, a];
+
+    let res = matrix![
+        1, 2, 1, 2;
+        3, 4, 3, 4;
+    ];
+
+    assert_eq_and_type!(m, res);
+}
+
+#[test]
+fn stack_single_col() {
+    let a = matrix![1, 2; 3, 4];
+    let m = stack![a; a];
+
+    let res = matrix![
+        1, 2;
+        3, 4;
+        1, 2;
+        3, 4;
+    ];
+
+    assert_eq_and_type!(m, res);
+}
+
+#[test]
+#[rustfmt::skip]
+fn stack_expr() {
+    let a = matrix![1, 2; 3, 4];
+    let b = matrix![5, 6; 7, 8];
+    let m = stack![a + b; 2i32 * b - a];
+
+    let res = matrix![
+         6,  8;
+        10, 12;
+         9, 10;
+        11, 12;
+    ];
+
+    assert_eq_and_type!(m, res);
+}
+
+#[test]
+fn stack_edge_cases() {
+    {
+        // Empty stack should return zero matrix with specified type
+        let _: SMatrix<i32, 0, 0> = stack![];
+        let _: SMatrix<f64, 0, 0> = stack![];
+    }
+
+    {
+        // Case suggested by @tpdickso: https://github.com/dimforge/nalgebra/pull/1080#discussion_r1435871752
+        let a = matrix![1, 2;
+                        3, 4];
+        let b = DMatrix::from_data(VecStorage::new(Dyn(2), Dyn(0), vec![]));
+        assert_eq!(
+            stack![a, 0;
+                          0, b],
+            matrix![1, 2;
+                           3, 4;
+                           0, 0;
+                           0, 0]
```

Comment:
`rustfmt` might have done something odd here, putting the first row of entries on a newline from the opening bracket might fix it!

Reply:
Good catch!